File: Portfolio/Portfolio.Blazor/Data/Entities/ReferenceCollectionTranslation.cs
using Portfolio.Blazor.Data;

namespace Portfolio.Blazor.Data.Entities;

public sealed class ReferenceCollectionTranslation
{
    public int Id { get; set; }
    public int ReferenceCollectionId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string? Intro { get; set; }

    [JsonOrEmpty]
    public string? Seo { get; set; }

    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public ReferenceCollection? ReferenceCollection { get; set; }
}

File: src/laravel/app/Models/ReferenceCollectionTranslation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ReferenceCollectionTranslation extends Model
{
    use HasFactory;

    protected $fillable = ['reference_collection_id', 'locale', 'title', 'description', 'intro', 'seo'];

    protected $casts = ['seo' => 'array'];

    /**
     * @return BelongsTo<ReferenceCollection, $this>
     */
    public function referenceCollection(): BelongsTo
    {
        return $this->belongsTo(ReferenceCollection::class);
    }
}
